codegen/lib/graphql_java_gen.rb

```diff
@@ -274,4 +274,34 @@ def abstract_types
       end
     end
   end
+
+  def java_doc(element)
+    doc = ''
+    unless element.description.nil?
+      description = wrap_text(element.description, 100)
+      description = format_java_doc_list(description)
+      description = description.chomp("\n").gsub("\n", "\n* ")
+      doc << "/**\n"
+      doc << '* '
+      doc << description
+      doc << "\n*/"
+    end
+    doc
+  end
+
+  def wrap_text(text, col_width=80)
+    text.gsub!( /(\S{#{col_width}})(?=\S)/, '\1 ' )
+    text.gsub!( /(.{1,#{col_width}})(?:\s+|$)/, "\\1\n" )
+    text
+  end
+
+  def format_java_doc_list(text)
+    parts = text.split(':')
+    list_pattern = /\s*?-\s*?`(.+?)`\s*?/
+    if parts.count > 1 && list_pattern.match(parts.last)
```

Comment:
This is really weird.  It is assuming that there is a single list in the description and that it starts with `:`.  Shouldn't we be looking for a markdown list rather than using heuristics like this?